docs/core/settings/syntax.md

```diff
@@ -447,3 +452,7 @@ Note that these are also Case Sensitive.
 
 These can also be used for external config files, but you need to list them in
 [[setting,import_environment]] so that processes can see them.
+
+If you use `${env:}` instead of `$ENV`, this will cause the environment variable(s) to be
+evaluated later. This can cause problems with files since config process cannot open
+them anymore. It is recommended to use `$ENV` instead when possible.
```

Comment:
* I think you mean `%{env:}`, not $
 * It's not clear it's talking about using %env with file type settings.
 * If using %env with file setting types, it's not config process that opens them, so it's wrong to say config process cannot open them.

Reply:
Maybe: `For [[link,settings_types_file]] it better to use $ENV: instead of %{env:}. This is because with $ENV: the expansion and file reading is done by the config process, which normally runs as root. With %{env:} the expansion is delayed until the process accessing the setting expands it, and the process may not have enough permissions to open the file.`